hkdf/src/hkdf.rs

```diff
@@ -27,19 +27,26 @@ impl<D> Hkdf<D>
           D::OutputSize: ArrayLength<u8>
 {
     #[deprecated(since="0.4.0", note="renamed to extract, with arguments in reversed order")]
-    pub fn new(ikm: &[u8], salt: &[u8]) -> Hkdf<D> {
+    pub fn new(ikm: &[u8], salt: Option<&[u8]>) -> Hkdf<D> {
         Self::extract(salt, ikm)
     }
 
     /// The RFC5869 HKDF-Extract operation
-    pub fn extract(salt: &[u8], ikm: &[u8]) -> Hkdf<D> {
+    pub fn extract(salt: Option<&[u8]>, ikm: &[u8]) -> Hkdf<D> {
         // The hmac-0.5 MAC trait (which provides new()) is now defined to
         // return a Result, apparently to support things like CMAC which
         // require a specific key length. As far as I can tell, HMAC in
```

Comment:
We can remove this comment and replace it with `.expect("HMAC can accept keys of any size")`.